build.rs

```diff
@@ -1,9 +1,6 @@
 fn main() -> anyhow::Result<()> {
     if std::env::var("TARGET").unwrap() == "xtensa-esp32-espidf" {
-        embuild::build::CfgArgs::output_propagated("ESP_IDF")?;
-        embuild::build::LinkArgs::output_propagated("ESP_IDF")?;
-        let cfg = embuild::build::CfgArgs::try_from_env("ESP_IDF")?;
-        cfg.output();
+        embuild::espidf::sysenv::output();
```

Comment:
Confidence in this change is ... uncertain. I'm taking this from what the esp-idf template does: https://github.com/esp-rs/esp-idf-template/blob/master/cargo/build.rs#L2.

See below for my comment on where things currently break. Perhaps more of what is done in https://github.com/ivmarkov/rust-esp32-std-demo/blob/main/build.rs is required?

Reply:
Using `embuild::espidf::sysenv::output()` with an updated `embuild` seems to necessary, so I'm keeping that change.

So far though, I haven't been able to remove the link error when I drop the `esp-idf-svc` dependency from `examples/Cargo.toml`. So, I'm going to restore that line for now so we can lock in the gains here, and then investigate how we can remove it once this is merged.